tests/devices/test_preprocess.py

```diff
@@ -512,6 +517,263 @@ def test_error_incompatible_mcm_method(self):
             _, _ = mid_circuit_measurements(tape, dev, mcm_config)
 
 
+class TestMeasurementsFromCountsOrSamples:
```

Comment:
Is it a good idea to use shot mocker here? I feel maybe it's not necessary to check the numeric accuracy here since we focus on the transforms